ui/modal/controller.go

```diff
@@ -75,13 +83,46 @@ func (m ViewModel) HandleMessage(msg tea.Msg) (*ViewModel, tea.Cmd) {
 			// If the previous state is not active
 			if ok {
 				if !m.transactionModal.Active {
-					if acct.Participation != nil &&
-						acct.Participation.VoteFirstValid == m.transactionModal.Participation.Key.VoteFirstValid {
-						m.SetActive(true)
-						m.infoModal.Active = true
-						m.infoModal.Prefix = "Successfully registered online!\n"
-						m.HasPrefix = true
-						m.SetType(app.InfoModal)
+					if acct.Participation != nil {
+						// comparing values to detect corrupted/non-resident keys
+						fvMatch := boolToInt(acct.Participation.VoteFirstValid == m.transactionModal.Participation.Key.VoteFirstValid)
+						lvMatch := boolToInt(acct.Participation.VoteLastValid == m.transactionModal.Participation.Key.VoteLastValid)
+						kdMatch := boolToInt(acct.Participation.VoteKeyDilution == m.transactionModal.Participation.Key.VoteKeyDilution)
+						selMatch := boolToInt(bytes.Equal(acct.Participation.SelectionParticipationKey, m.transactionModal.Participation.Key.SelectionParticipationKey))
+						votMatch := boolToInt(bytes.Equal(acct.Participation.VoteParticipationKey, m.transactionModal.Participation.Key.VoteParticipationKey))
+						spkMatch := boolToInt(bytes.Equal(*acct.Participation.StateProofKey, *m.transactionModal.Participation.Key.StateProofKey))
+						matchCount := fvMatch + lvMatch + kdMatch + selMatch + votMatch + spkMatch
+						if matchCount == 6 {
+							m.SetActive(true)
+							m.infoModal.Active = true
+							m.infoModal.Prefix = "Successfully registered online!\n"
+							m.HasPrefix = true
+							m.SetType(app.InfoModal)
+						} else if matchCount >= 4 {
```

Comment:
It may be safer to just have a binary here. Not sure of the use case where only 2 of the keys are not valid and we want to continue.